libcnb-test/src/test_runner.rs

```diff
@@ -160,11 +159,42 @@ impl TestRunner {
         let test_context = TestContext {
             pack_stdout: output.stdout,
             pack_stderr: output.stderr,
-            image_name,
+            docker_resources,
             config: config.clone(),
             runner: self,
         };
 
         f(test_context);
     }
 }
+
+pub(crate) struct TemporaryDockerResources {
+    pub(crate) build_cache_volume_name: String,
+    pub(crate) image_name: String,
+    pub(crate) launch_cache_volume_name: String,
+}
+
+impl TemporaryDockerResources {
+    pub fn new() -> Self {
```

Comment:
I think naming this function `new` is potentially confusing. I read the usage of it first (since it comes before this in the PR diff) and was wondering what would happen. I could see it either initialising it with empty-_like_ values, random names (like seen here) and actually side-effecting to do some sort of initialisation with the Docker process (and wondering what that even would be).

Finding a great alternative name for this function isn't easy - I tried to come up with a good suggestion but nothing came out of that.

However, we only create a new value of this struct _once_. This leaves us with the option of moving the body of this `new` function to the place where it's used and not giving it a name.  This has the advantage that the fact that only random names are generated and no side-effects are happening is absolutely clear at the call-site.

Reply:
Agree - changed in d11167bf4b342144e51332b21f3923fea0bc6736 :-)